mock-app-verifier/programs/verifier/src/lib.rs

```diff
@@ -81,26 +79,22 @@ pub mod mock_verifier {
     }
 
     pub fn light_instruction_second<'a, 'b, 'c, 'info>(
-        ctx: Context<'a, 'b, 'c, 'info, LightInstructionSecond<'info>>,
+        ctx: Context<'a, 'b, 'c, 'info, LightInstructionSecond<'info, 2>>,
         inputs: Vec<u8>,
     ) -> Result<()> {
-        let _ = inputs
-            .chunks(32)
-            .map(|input| {
-                ctx.accounts
-                    .verifier_state
-                    .checked_public_inputs
-                    .push(input.to_vec())
-            })
-            .collect::<Vec<_>>();
+        inputs.chunks(32).enumerate().for_each(|(i, input)| {
```

Comment:
What's the difference between map and for_each?
Why do you prefer the latter in this case?
(happy with your changes just curious)

Reply:
Closures inside `map` are supposed to return some value on each iteration. Then `collect` is used to make some data structure from those values.
`for_each` is just performing actions given in the closure without returning anything.

Here, we just want to add stuff to an array, we don't want to collect anything. The previous code, which was `collect`ing a `Vec<_>` was unncesessary. In fact, the resulting collected type was most likely `Vec<()>` and then it was ignored anyway with `let _ =`.

Reply:
thanks!